modules/docs/arrow-docs/docs/docs/generic/coproduct/README.md

```diff
@@ -0,0 +1,193 @@
+                val registeredCar = registration.car
+
+                showCarSuccessfullyRegistered(registeredCar)
+            }
+    )
+```
+
+This example returns `Unit` because all of these are side effects, let's say our application was built for a command line and we just have to show a `String` for the result of the call (if only it was always that easy):
+
+```kotlin:ank
+import arrow.generic.*
+import arrow.generic.coproduct3.Coproduct3
+import arrow.generic.coproduct3.fold
+
+fun renderApiResult(apiResult: Coproduct3<CommonServerError, RegistrationError, Registration>): String = apiResult.fold(
+            { commonError ->
+                when (commonError) {
+                    ServerError -> "Server error, try again later."
+                    UserUnauthorized -> "Unauthorized!"
+                    OverRequestLimit -> "Too many api requests, try again later."
+                }
+            },
+            { registrationError ->
+                when (registrationError) {
+                    CarAlreadyRegistered -> "Car already registered."
+                    StolenCar -> "Car reported stolen!"
+                }
+            },
+            { registration ->
+                "Successfully Registered: $registration.car"
+            }
+    )
+```
+
+Here we're able to return the result of the `fold` and since it's exhaustively evaluated, we're forced to handle all cases! Neat! Let's say we also want to store the `Registration` object into our database when we successfully register a car. We don't really want to have to `fold` over every single case just to handle something for the `Registration`, this is where `select<T>` comes to the rescue!
+
+##### select
+
+We're able to take a Coproduct and `select` the type we care about from it. `select` returns an `Option`, if the value of the Coproduct was for the type you're trying to `select`, you'll get `Some`, if it was not the type used with `select`, you'll get `None`.
+
+```kotlin:ank
+import arrow.generic.*
+import arrow.generic.coproduct3.Coproduct3
+import arrow.generic.coproduct3.select
+
+fun handleApiResult(
+        database: Database,
+        apiResult: Coproduct3<CommonServerError, RegistrationError, Registration>
+): Unit {
+    apiResult.select<Registration>()
+            .fold(
+                    {}, //Wasn't Registration, nothing to do here
+                    { database.insertRegistration(it) }
+            )
+}
+```
+
+`select` can only be called with a type that exists on the Coproduct, if the type doesn't exist, it won't compile:
+```kotlin:ank
+import arrow.generic.*
+import arrow.generic.coproduct3.Coproduct3
+import arrow.generic.coproduct3.select
+
+fun handleApiResult(apiResult: Coproduct3<CommonServerError, RegistrationError, Registration>): Unit {
+//    apiResult.select<String>()
+//error: type mismatch: inferred type is Coproduct3<CommonServerError, RegistrationError, Registration> but Coproduct3<String, *, *> was expected
```

Comment:
This reminds me that Ank should have a mode for expecting and printing errors similar to how we do now `ank:silent`. but `ank:error` so users are not forced to comment out code that fails to compile and can let the compiler provide the actual error that gets rendered along the snippet.

Reply:
/cc @Guardiola31337